src/pudl/output/pudltabl.py

```diff
@@ -1097,15 +1104,33 @@ def mcoe(
                 min_cap_fact=min_cap_fact,
                 max_cap_fact=max_cap_fact,
                 all_gens=all_gens,
+                extra_gens_cols=extra_gens_cols,
+                all_gens_cols=all_gens_cols,
             )
         return self._dfs["mcoe"]
 
-    def gens_mega_eia(self, update: bool = False) -> pd.DataFrame:
+    def gens_mega_eia(
+        self,
+        update: bool = False,
+        extra_gens_cols=[
```

Comment:
Using a (mutable) `list` [as a default argument is dangerous](https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/), since if the variable is modified in the function, it remains modified for subsequent calls to the function, which can result in unexpected side effects. The recommended solution is to use `None` as the default, check for that value and assign the default value inside the function. You could use an immutable sequence for the default instead, like a `tuple`.

Having two different parameters which define the list of generator attributes to include makes the behavior of the function ambiguous.  What happens if you say `all_gens_cols=True` but also give it set of `extra_gens_cols`?  One solution to this could be to accept either a special literal like `all` or a list of specific columns for `extra_gens_cols` and eliminate the `all_gens_cols` parameter.

Also let's try and add type hints whenever we're touching existing functions or creating new ones.

These comments would also apply to the `mcoe()` function and `plant_parts_eia()` below.

Reply:
Good catch, I've never run into this issue of using mutable types as default arguments. Good to know. 

I think a special literal for `extra_gens_cols` to get rid of `all_gens_cols` makes the most sense.